diff --git a/index.d.ts b/index.d.ts
index 49e4f966198541e1ebc4029c3924d6fcd2bc7173..d31b0c7420acf4807eba6c96c7b50e80e81273b5 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -4,6 +4,30 @@ type Options = {
 	edge?: boolean;
 	external?: string[];
 	split?: boolean;
+	platform?: VercelConfig;
 };
 
+type ImageFormat = 'image/avif' | 'image/webp';
+
+type RemotePattern = {
+  protocol?: 'http' | 'https';
+  hostname: string;
+  port?: string;
+  pathname?: string;
+};
+
+type ImagesConfig = {
+  sizes: number[];
+  domains: string[];
+  remotePatterns?: RemotePattern[];
+  minimumCacheTTL?: number; // seconds
+  formats?: ImageFormat[];
+  dangerouslyAllowSVG?: boolean;
+  contentSecurityPolicy?: string;
+};
+
+type VercelConfig = {
+  images?: ImagesConfig,
+}
+
 export default function plugin(options?: Options): Adapter;
diff --git a/index.js b/index.js
index 529225d7cf28f9aaf8139b350c60b5db0f868f7f..843cffc49549b02af708af11bfbb397e23a1eefa 100644
--- a/index.js
+++ b/index.js
@@ -5,7 +5,7 @@ import { nodeFileTrace } from '@vercel/nft';
 import esbuild from 'esbuild';
 
 /** @type {import('.').default} **/
-const plugin = function ({ external = [], edge, split } = {}) {
+const plugin = function ({ external = [], edge, split, platform } = {}) {
 	return {
 		name: '@sveltejs/adapter-vercel',
 
@@ -134,6 +134,10 @@ const plugin = function ({ external = [], edge, split } = {}) {
 				await generate_function('render', '/.*', builder.generateManifest);
 			}
 
+			if (platform && platform?.images) {
+				config.images = platform.images;
+			}
+
 			builder.log.minor('Copying assets...');
 
 			builder.writeClient(dirs.static);
@@ -219,7 +223,8 @@ function static_vercel_config(builder) {
 				handle: 'filesystem'
 			}
 		],
-		overrides
+		overrides,
+		images: {},
 	};
 }
 